operator/pkg/client/client_factory.go

```diff
@@ -374,3 +385,22 @@ func setKubernetesDefaults(config *rest.Config) error {
 	}
 	return rest.SetKubernetesDefaults(config)
 }
+
+func (s *SingletonClients) checkAndResetMapper(obj runtime.Object) (*meta.RESTMapping, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	mapping, err := s.discoveryShortcutExpander.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if gvk.Empty() {
+		return mapping, nil
+	}
+	if err != nil {
+		if meta.IsNoMatchError(err) {
+			s.discoveryRESTMapper.Reset()
```

Comment:
If you do a reset, you do not need to error out. You can simply recall `.RESTMapping` and then error out if `err != nil` after the second call.

Also a method called `checkAndResetMapper` should not return a Mapping imo.

You could call it `ToRESTMapping` instead